azurerm/provider_test.go

```diff
@@ -49,6 +52,21 @@ func testAccPreCheck(t *testing.T) {
 			t.Fatalf("`%s` must be set for acceptance tests!", variable)
 		}
 	}
+
+	// If "TF_PARL" is set to some non-empty value, all the parallel test cases
+	// will be run in parallel. It requires that the test cases are all
+	// independent ones without any shared resource before turn it on.
```

Comment:
I'd argue the second sentence in this comment is incorrect since it's not the reason why we can't run these tests in Parallel - which are mostly due to limitations in Azure (either Rate Limiting, or Usage Limits [such as only a single instance in a region w/Network Watcher])

Reply:
Agree, updated the comments and the code for better reading.